galsim/sensor.py

```diff
@@ -0,0 +1,182 @@
+#    and/or other materials provided with the distribution.
+"""@file sensor.py
+
+The Sensor classes implement the process of turning a set of photons incident at the surface
+of the detector in the focal plane into an image with counts of electrons in each pixel.
+
+The Sensor class itself implements the simplest possible sensor model, which just converts each
+photon into an electron in whatever pixel is below the location where the photon hits.
+However, it also serves as a base class for other more classes that implement more sophisticated
+treatments of the photon to electron conversion and the drift from the conversion layer to the
+bottom of the detector.
+"""
+
+import numpy as np
+import galsim
+import glob
+import os
+
+class Sensor(object):
+    """
+    The base class for other sensor models, and also an implementation of the simplest possible
+    sensor model that just converts each photon into an electron and drops it in the appropriate
+    pixel.
+    """
+    def __init__(self):
+        pass
+
+    def accumulate(self, photons, image):
+        """Accumulate the photons incident at the surface of the sensor into the appropriate
+        pixels in the image.
+
+        Each photon has a position, which corresponds to the (x,y) position at the top of the
+        sensor.  In general, they may also have incidence directions and wavelengths, although
+        these are not used by the base class implementation.
+
+        The base class implementation simply accumulates the photons above each pixel into that
+        pixel.
+
+        @param photons      A PhotonArray instance describing the incident photons
+        @param image        The image into which the photons should be accumuated.
+        """
+        return photons.addTo(image.image)
+
+
+class SiliconSensor(Sensor):
+    """
+    A model of a silicon-based CCD sensor that converts photons to electrons at a wavelength-
+    dependent depth (probabilistically) and drifts them down to the wells, properly taking
+    into account the repulsion of previously accumulated electrons (known as the brighter-fatter
+    effect).
+
+    @param dir              The name of the directory which contains the sensor information,
+                            presumably calculated from the Poisson_CCD simulator.  This directory
+                            may be specified either as an absolute path or as a subdirectory of
+                            share_dir/sensors/, where share_dir is `galsim.meta_data.share_dir`.
+                            It must contain, at a minimum, the *.cfg file used to simulate the
+                            pixel distortions, and the *_Vertices.dat file which carries the
+                            distorted pixel information.  [default: 'lsst_itl']
+    @param strength         Set the strength of the brighter-fatter effect relative to the
+                            amount in specified by the Poisson simulation results.  [default: 1]
```

Comment:
I switched the num_elec parameter to a strength parameter, since you have the reference number of electrons stored in the config file now, so the only purpose left for this was to change the strength of the effect.  I think it will be easier for users to just enter `strength=2` if they want to double the strength of the effect relative to the simulator results.  This is equivalent to setting `num_elec = config['CollectedCharge_0_0'] / 2` in the old scheme.